websock/session.nim

```diff
@@ -18,6 +18,38 @@ import pkg/chronos/streams/asyncstream
 logScope:
   topics = "websock ws-session"
 
+proc updateReadMode(ws: WSSession): bool =
```

Comment:
Switching from text to binary or binary to text in a single message are not allowed in the spec. `Opcode.Text` or `Opcode.Binary` should always the be the first frame in a message. In a multi frames message, 2nd frame to final frame should always `Opcode.Cont`.

FYI: the autobahn test result need manual inspection because it will never fail the CI with current setup regardless of failing cases.

Reply:
This was mentioned in the forst item of the introductory comment. The internal state machine _might_ allow for that. So in a defensive manner, an exception is thrown if a state change would not make sense.

Reply:
Re Autobahn: there is no point going there yet until partial frames can be processed reliably (see the new test in testwebsockets.nim.)